lib/streamy/helpers/rspec_helper.rb

```
@@ -17,8 +17,9 @@ def expect_event(topic: kind_of(String), priority: kind_of(Symbol), key: kind_of
         )
       end
 
-      def expect_no_event(type)
-        expect(streamy_deliveries).not_to include(a_hash_including(type: type.to_s))
+      def expect_no_event(type:)
```

any reason for this signature change?

I see that we have variations of using this `expect_no_event` in our codebase, I'm not sure if it's a mistake or intentional

```ruby
expect_no_event(:redacted_type)

expect_no_event(type: "redacted_type")

expect_no_event(topic: "redacted.test.object.avro", type: "redacted_type")

expect_no_event("unpublished_recipe")
```

since we're changing the signature, what do you think about making the method more flexible to accepts more than `type:`?

---

> any reason for this signature change?

Yes. The purpose of this PR is to fix the spec helper and to prevent false positives. Switching to a named parameter will help prevent incorrect usages in the future by not silently converting implicit hash parameter lists such as `expect_no_event(topic: "redacted.test.object.avro", type: "redacted_type")` to the single parameter `type` which then would not be found in the sent events because the helper was expecting a single value for type and also not looking in the correct location.

All that we currently need to effectively verify that an event is not present is the event `type:` value to look for. If, in the future, we have a clear need to validate additional values, we could modify the helper to do so. I do not see a need to implement support for anything else at this time.

As indicated in a comment above, I have [a follow on PR](https://github.com/cookpad/global-web/pull/16377) that will correct the usage of `expect_no_event` in global-web.